### PR TITLE
Victor/op 41 implement and test summarization of forum threads

### PR DESCRIPTION
**Pull request by victorsouza89**
implemented summarization of forum posts. added three main documents, all in `notebooks/004_summarization_exploring`. they are:

- `summarize.py` : is a streamlit notebook that allowed testing some different approaches and different scenarios.
- `util.py` :  stored some important functions to define models and workflow.
- `summarize_cli.py` :  standalone script that receives a forum thread url and summarizes it. allows choosing the llm model, default is gpt-4o. at this application i didn't notice any remarkable improvement by using claude.

### CODE REVIEW

File: data/003-snapshot-spaces-proposals-20240711/dataset.jsonl
Comment (ribeirojose) [inline data/003-snapshot-spaces-proposals-20240711/dataset.jsonl:1]: have you added anything to this dataset?

Comment (victorsouza89) [reply data/003-snapshot-spaces-proposals-20240711/dataset.jsonl:1]: I didn't. I guess something went weird when I merged into this branch the commit in which you added this dataset

File: notebooks/004_summarization_exploring/summarize.py [hunk 1]
```diff
@@ -81,8 +81,14 @@
         announcement_example = "https://gov.optimism.io/t/retro-funding-4-onchain-builders-round-details/7988"
         defalut_threads.append(announcement_example)
     if st.checkbox("Add Discussion Example"):
-        discussion_example = "https://gov.optimism.io/t/the-future-of-optimism-governance/6471"
+        discussion_example = "https://gov.optimism.io/t/how-to-start-a-project-at-optimism/7220"
         defalut_threads.append(discussion_example)
```
Comment (ribeirojose) [inline notebooks/004_summarization_exploring/summarize.py:85]:
```suggestion
        default_threads.append(discussion_example)
```

File: pkg/op-chat-brains/op_chat_brains/summarizer/summarizer.py [hunk 2]
```diff
@@ -7,8 +7,7 @@
 from op_chat_brains.config import FORUM_PATH, SNAPSHOT_PATH
 from op_chat_brains.documents.optimism import ForumPostsProcessingStrategy
 from op_chat_brains.exceptions import OpChatBrainsException
-from op_chat_brains.summarizer.internal_dialogue import InternalDialogue
```
Comment (ribeirojose) [inline pkg/op-chat-brains/op_chat_brains/summarizer/summarizer.py]: should we just delete this internal dialogue then?

**Review by ribeirojose — APPROVED** [review]
looking promising! 👌

**Pull request merged**